cvmfs/sync_union_tarball.cc

```diff
@@ -336,6 +338,7 @@ void SyncUnionTarball::CreateDirectories(const std::string &target) {
       new SyncItemDummyDir(dirname, filename, this, kItemDir));
 
   ProcessUnmaterializedDirectory(dummy);
+  dirs_[target] = dummy;
```

Comment:
Is this change related to the creation of `.cvmfscatalog`?

Reply:
Yes, it is the one access on L153 (https://github.com/cvmfs/cvmfs/pull/2284/files#diff-a076033a7bf9d98a9175b0dce22c8c8cR153).

Reply:
We are allowing the code to create catalogs also in directory that are not in the tar file, like the root directory where we extract the tar itself.

@jblomer